include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -91,7 +91,7 @@ struct range_iterator_end<Range, open>
 
 
 template <typename Range, typename Value, typename Reference = Value>
-class range_segment_iterator
+struct range_segment_iterator
```

Comment:
Fix done.
Also fixed a bug that was introduced by the usage of `boost::size` to initialize the `m_has_less_than_two_elements` member variable.
See PR #203.